apps/builder/app/routes/rest.patch.ts

```diff
@@ -216,15 +235,12 @@ export const action = async ({ request }: ActionArgs) => {
     dbBuildData.styles = serializeStyles(buildData.styles);
   }
 
-  // check any build data is changed because only assets could change
-  if (Object.keys(dbBuildData).length > 0) {
-    await prisma.build.update({
-      data: dbBuildData,
-      where: {
-        id_projectId: { projectId, id: buildId },
-      },
-    });
-  }
+  await prisma.build.update({
+    data: dbBuildData,
+    where: {
+      id_projectId: { projectId, id: buildId },
```

Comment:
See you read serverVersion few async commands above, everything here is not inside transaction.
So race conditions are possible.
To avoid you must add version check into update command too.

```
where: {
   ....
   version: serverVersion
}
```

And gracefully catch error in case of race condition